internal/app/admin/http/cache.go

```diff
@@ -151,31 +151,25 @@ func cacheDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		cacheKey := getParam(req.URL.Path)
 		c := orchestrator.Orchestrator.GetReadOnlyCache(*o)
-		var keysToPrint []string
-
-		// If wildcard suffix provided, output all cache entries that match given prefix.
-		// If no key is provided, output the entire cache.
-		if hasWildcardSuffix(cacheKey) {
-			// Retrieve all keys
-			allKeys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				fmt.Fprintf(w, "error in getting cache keys: %s", err.Error())
-				return
-			}
+		keysToPrint, err := getRelevantKeys(o, cacheKey, w)
+		if err == nil {
+			printCacheEntries(keysToPrint, c, w)
+		}
+	}
+}
 
-			// Find keys that match prefix of wildcard
-			rootCacheKeyName := strings.TrimSuffix(cacheKey, "*")
-			for potentialMatchKey := range allKeys {
-				if strings.HasPrefix(potentialMatchKey, rootCacheKeyName) {
-					keysToPrint = append(keysToPrint, potentialMatchKey)
-				}
+func clearCacheHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		if req.Method == "POST" {
```

Comment:
may be use a switch case.

Reply:
Updated it to use `http.MethodPost`, but since it is just a single case to check, I think a switch case isn't really necessary. Instead, I modified it to avoid using an else case. This fxn. was based of the [logHandler](https://github.com/envoyproxy/xds-relay/blob/master/internal/app/admin/http/handler.go#L152-L176), so I can fix the nits of that function in another PR.